### PR TITLE
Consolidate before/after hooks

### PR DESCRIPTION
**Pull request by timleslie**
Consolidate the beforeChange/beforeDelete hooks into beforeOperation, and afterChange/afterDelete hooks into afterOperation.

### CODE REVIEW

File: packages/keystone/src/lib/core/mutations/create-update.ts [hunk 1]
```diff
@@ -52,11 +52,11 @@ async function createSingle(
     runWithPrisma(context, list, model => model.create({ data }))
   );
 
-  return { item, afterChange };
+  return { item, afterOperation };
 }
 
 export class NestedMutationState {
-  #afterChanges: (() => void | Promise<void>)[] = [];
+  #afterOpertions: (() => void | Promise<void>)[] = [];
```
Comment (mitchellhamilton) [inline packages/keystone/src/lib/core/mutations/create-update.ts:59]:
```suggestion
  #afterOperations: (() => void | Promise<void>)[] = [];
```

File: packages/keystone/src/lib/core/mutations/create-update.ts [hunk 2]
```diff
@@ -68,20 +68,20 @@ export class NestedMutationState {
...
+    const { item, afterOperation } = await createSingle(
       { data },
       list,
       context,
       operationAccess,
       writeLimit
     );
 
-    this.#afterChanges.push(() => afterChange(item));
+    this.#afterOpertions.push(() => afterOperation(item));
```
Comment (mitchellhamilton) [inline packages/keystone/src/lib/core/mutations/create-update.ts:79]:
```suggestion
    this.#afterOperations.push(() => afterOperation(item));
```

File: packages/keystone/src/lib/core/mutations/create-update.ts [hunk 3]
```diff
@@ -68,20 +68,20 @@ export class NestedMutationState {
...
     return { id: item.id as IdType };
   }
 
-  async afterChange() {
-    await promiseAllRejectWithAllErrors(this.#afterChanges.map(async x => x()));
+  async afterOperation() {
+    await promiseAllRejectWithAllErrors(this.#afterOpertions.map(async x => x()));
```
Comment (mitchellhamilton) [inline packages/keystone/src/lib/core/mutations/create-update.ts:84]:
```suggestion
    await promiseAllRejectWithAllErrors(this.#afterOperations.map(async x => x()));
```

**Review by emmatown — CHANGES_REQUESTED** [review]
Can you make the inputData and resolvedData `undefined` when it doesn't exist rather than `null`?

**Review by emmatown — APPROVED** [review]

**Pull request merged**